doc/changelog/previous-versions/v3.x.md

```diff
@@ -0,0 +1,1271 @@
+Feat: Add Codacy badge
+Fix(Core/Quest): Correct add locale in quest
+Fix(Core/DB/Ulduar): Assembly of Iron
+Fix(DB/NPC): Librarian Garren
+Fix(DB/Quest): Ogre Head On A Stick
+Fix(Core/Player): fix periodic eating and drinking emotes
+refactor(DB/world): graveyard_zone
+Fix(DB): Fixing several Npc issues
+DB restructuring - world - achievement_reward
+fix(DB/Quest): WANTED: Boss Grog'ak
+Skill perfect item template
+Switch to C++14 to solve compilation issues
+Feat:Update README.md
+Fix(Core/Common): Added header <unordered_map> in file DBCStore.h
+Fix(Core/Ticket System): Ticket comment visual output bug
+feat(Core/Chat): Add config option to allow use LFG anywhere not only in city
+fix(DB): Fjord Hawk, Fjord Turkey
+fix(DB/Core): play event after quest "Hero of the Mag'har"; extend SAI to play music
+fix(DB): a few improvements to the giant wandering outdoor creatures
+fix(DB): zeppelin master - remove gossip option "Where is the zeppelin now?"
+Fix(Core): GM command not working while GM mode is on
+feat(CI): improve Travis build time
+Fix(Scripts/Instances/Oculus): improvements
+fix(DB): Malmortis creature text
+fix(Core/WorldSession): change how to storing data in account_tutorial
+Fix(Scripts/Spells): Check range to fire totem when casting Fire Nova.
+fix(Core/Player): close all character tickets when delete
+fix(Core/Commands): fix reload creature_template and creature movment
+fix(Modules): solve breaking change
+fix(core): prevent unsupported-by-client passwords
+feat(core): allow to debug KickPlayer()
+feat(cpp): enable C++17 support
+fix(Docker/macOS): solve disconnection issues
+feat(core): allow to disable auto-kick for idle players
+Fix(Core/Ulduar): Multiple fixes to Flame Leviathan
+feat(Core/Movement): Implement move time skipped handler
+Eluna: load the configuration file using the new method.
+fix(Docker): no need to duplicate *.dist files
+feat(modules): Allow modules to extend Battlegrounds
+fix(DB/creature): correct phaseMask of NPC Ravenous jormungar
+fix(Core/DB/Quest): Battle for the Undercity
+Core/ChatHandler: Update Addon message read fix (#811)
+fix(DB): Pit of Saron intro
+feat(CI): add commit hashes to the automatic SQL import
+chore(readme): fix stackoverflow default view
+feat(Commands): Added cheat commands
+feat(docker): add cache for faster rebuild
+fix(DB/Item): Hateful Gladiator's Longbow No Range
+feat(CI): dynamically calculate thread count for build process
+Core/Skills: Add FishingBaseSkillLevel dynamic change method
+fix(DB/Quest): Morrowgrain to darnassus
+style(cpp): DBC AchievementCriteriaTypes indentation
+fix(CI): break on errors in CI scripts
+Core/Eluna: Hotfix call sEluna->OnGameEventStart method arg register
+feat(CI): improve Travis configuration
+Core/Hooks: Adds hooks for start and stop game events
+Core/Hooks: Add new Player hooks
+chore(docs): use new wiki
+fix(CI): Travis triggers the AC bot only once
+fix(DB): sql update chain for all DBs
+DB: Fixing some issues
+```
```

Comment:
not sure if we should put the full changelog here, as we are deciding to only add breaking changes in the future changelogs... so it might look inconsistent.

Reply:
We need them as an example of where to put the changelogs of the previous releases and how they should be named.
However, how to write a changelog implies that you follow the wiki guidelines and it's written also into the file that is automatically generated. People should use it as they already do with the SQL in-pending files. You first have to read the wiki otherwise you do not know how to do that just by copying other sql.